src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -7,171 +7,193 @@
+        protected override void ValidatePropertyType(DiagnosticBag diagnostics)
+        {
+            base.ValidatePropertyType(diagnostics);
+            VerifyOverridesEqualityContractFromBase(this, diagnostics);
+        }
 
-            internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+        internal static void VerifyOverridesEqualityContractFromBase(PropertySymbol overriding, DiagnosticBag diagnostics)
+        {
+            if (overriding.ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+            {
+                return;
+            }
 
-            internal override bool GenerateDebugInfo => false;
+            bool reportAnError = false;
 
-            public override DllImportData? GetDllImportData() => null;
+            if (!overriding.IsOverride)
+            {
+                reportAnError = true;
+            }
+            else
+            {
+                var overridden = overriding.OverriddenProperty;
 
-            internal override ImmutableArray<string> GetAppliedConditionalSymbols()
-                => ImmutableArray<string>.Empty;
+                if (overridden is object &&
+                    !overridden.ContainingType.Equals(overriding.ContainingType.BaseTypeNoUseSiteDiagnostics, TypeCompareKind.AllIgnoreOptions))
+                {
+                    reportAnError = true;
+                }
+            }
 
-            internal override IEnumerable<SecurityAttribute> GetSecurityInformation()
-                => Array.Empty<SecurityAttribute>();
+            if (reportAnError)
+            {
+                diagnostics.Add(ErrorCode.ERR_DoesNotOverrideBaseEqualityContract, overriding.Locations[0], overriding, overriding.ContainingType.BaseTypeNoUseSiteDiagnostics);
+            }
+        }
 
-            internal override bool IsMetadataNewSlot(bool ignoreInterfaceImplementationChanges = false) => !IsOverride;
+        internal sealed class GetAccessorSymbol : SourcePropertyAccessorSymbol
+        {
+            internal GetAccessorSymbol(
+                NamedTypeSymbol containingType,
+                string name,
+                SourcePropertySymbolBase property,
+                DeclarationModifiers propertyModifiers,
+                ImmutableArray<MethodSymbol> explicitInterfaceImplementations,
+                Location location,
+                CSharpSyntaxNode syntax,
+                DiagnosticBag diagnostics)
+                : base(
+                       containingType,
+                       name,
+                       property,
+                       propertyModifiers,
+                       explicitInterfaceImplementations,
+                       location,
+                       syntax,
+                       hasBody: false,
+                       hasExpressionBody: false,
+                       isIterator: false,
+                       modifiers: new SyntaxTokenList(),
```

Comment:
singleton here too?

Reply:
> singleton here too?

Same here, it is a struct

---
In reply to: [452481371](https://github.com/dotnet/roslyn/pull/45831#discussion_r452481371) [](ancestors = 452481371)